homeassistant/components/demo/lock.py

```diff
@@ -24,24 +33,67 @@ class DemoLock(LockEntity):
 
     _attr_should_poll = False
 
-    def __init__(self, name: str, state: str, openable: bool = False) -> None:
+    def __init__(
+        self,
+        name: str,
+        state: str,
+        openable: bool = False,
+        jam_on_operation: bool = False,
+    ) -> None:
         """Initialize the lock."""
         self._attr_name = name
-        self._attr_is_locked = state == STATE_LOCKED
         if openable:
             self._attr_supported_features = SUPPORT_OPEN
+        self._state = state
+        self._openable = openable
+        self._jam_on_operation = jam_on_operation
+
+    @property
+    def is_locking(self):
+        """Return true if lock is locking."""
+        return self._state == STATE_LOCKING
+
+    @property
+    def is_unlocking(self):
+        """Return true if lock is unlocking."""
+        return self._state == STATE_UNLOCKING
 
-    def lock(self, **kwargs):
+    @property
+    def is_jammed(self):
+        """Return true if lock is jammed."""
+        return self._state == STATE_JAMMED
+
+    @property
+    def is_locked(self):
+        """Return true if lock is locked."""
+        return self._state == STATE_LOCKED
+
+    async def async_lock(self, **kwargs):
         """Lock the device."""
-        self._attr_is_locked = True
-        self.schedule_update_ha_state()
+        self._state = STATE_LOCKING
+        self.async_write_ha_state()
+        await asyncio.sleep(2)
+        if self._jam_on_operation:
+            self._state = STATE_JAMMED
+        else:
+            self._state = STATE_LOCKED
+        self.async_write_ha_state()
 
-    def unlock(self, **kwargs):
+    async def async_unlock(self, **kwargs):
         """Unlock the device."""
-        self._attr_is_locked = False
-        self.schedule_update_ha_state()
+        self._state = STATE_UNLOCKING
+        self.async_write_ha_state()
+        await asyncio.sleep(2)
+        self._state = STATE_UNLOCKED
+        self.async_write_ha_state()
 
-    def open(self, **kwargs):
+    async def async_open(self, **kwargs):
         """Open the door latch."""
-        self._attr_is_locked = False
-        self.schedule_update_ha_state()
+        self._state = STATE_UNLOCKED
+        self.async_write_ha_state()
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        if self._openable:
+            return SUPPORT_OPEN
```

Comment:
Inconsistent return value. Please have all function exits return a value specifically when returning a value anywhere in the function.

Reply:
https://github.com/home-assistant/core/pull/54300